typescript/infra/config/environments/mainnet3/warp/configGetters/getRenzoPZETHWarpConfig.ts

```diff
@@ -31,26 +30,140 @@ const pzEthValidators = {
+    },
+  };
 
-const existingProxyAdmins: ChainMap<{ address: string; owner: string }> = {
-  ethereum: {
-    address: '0x4f4671Ce69c9af15e33eB7Cf6D1358d1B39Af3bF',
-    owner: '0xD1e6626310fD54Eceb5b9a51dA2eC329D6D4B68A',
-  },
-  zircuit: {
-    address: '0x8b789B4A56675240c9f0985B467752b870c75711',
-    owner: '0x8410927C286A38883BC23721e640F31D3E3E79F8',
-  },
+  const config = getEnvironmentConfig('mainnet3');
+  const multiProvider = await config.getMultiProvider();
+  const registry = await getMainnet3Registry();
+
+  const validatorDiff = symmetricDifference(
+    new Set(chainsToDeploy),
+    new Set(Object.keys(pzEthValidators)),
+  );
+  const safeDiff = symmetricDifference(
+    new Set(chainsToDeploy),
+    new Set(Object.keys(pzEthSafes)),
+  );
+  if (validatorDiff.size > 0) {
+    throw new Error(
+      `chainsToDeploy !== validatorConfig, diff is ${Array.from(
+        validatorDiff,
+      ).join(', ')}`,
+    );
+  }
+  if (safeDiff.size > 0) {
+    throw new Error(
+      `chainsToDeploy !== safeDiff, diff is ${Array.from(safeDiff).join(', ')}`,
+    );
+  }
+
+  const tokenConfig = Object.fromEntries<HypTokenRouterConfig>(
+    await Promise.all(
+      chainsToDeploy.map(
+        async (chain): Promise<[string, HypTokenRouterConfig]> => {
+          const addresses = await registry.getChainAddresses(chain);
+          assert(addresses, 'No addresses in Registry');
+          const { mailbox } = addresses;
+
+          const mailboxContract = Mailbox__factory.connect(
+            mailbox,
+            multiProvider.getProvider(chain),
+          );
+          const defaultHook = await mailboxContract.defaultHook();
+          const ret: [string, HypTokenRouterConfig] = [
+            chain,
+            {
+              isNft: false,
+              type:
+                chain === lockboxChain
+                  ? TokenType.XERC20Lockbox
+                  : TokenType.XERC20,
+              token: chain === lockboxChain ? lockbox : xERC20,
+              owner: pzEthSafes[chain],
+              gas: warpRouteOverheadGas,
+              mailbox,
+              interchainSecurityModule: {
+                type: IsmType.AGGREGATION,
+                threshold: 2,
+                modules: [
+                  {
+                    type: IsmType.ROUTING,
+                    owner: pzEthSafes[chain],
+                    domains: buildAggregationIsmConfigs(
+                      chain,
+                      chainsToDeploy,
+                      pzEthValidators,
+                    ),
+                  },
+                  {
+                    type: IsmType.FALLBACK_ROUTING,
+                    domains: {},
+                    owner: pzEthSafes[chain],
+                  },
+                ],
+              },
+              hook: getRenzoHook(defaultHook, chain),
+              proxyAdmin: existingProxyAdmins[chain],
+            },
+          ];
+
+          return ret;
+        },
+      ),
+    ),
+  );
```

Comment:
why is this changing so much?
can we have 1 PR just to setup staging with an identical config and a separate PR for value changes?